src/instructlab/model/convert.py

```diff
@@ -1,127 +1,115 @@
-
-    # compute model name from model-dir if not supplied
-    if model_name is None:
-        mlx_q_suffix = "-mlx-q"
-        model_name = model_dir.split("/")[-1]
-        model_name = model_name.replace(mlx_q_suffix, "")
-
-    if adapter_file is None:
-        adapter_file = os.path.join(model_dir, "adapters.npz")
-    source_model_dir = model_dir
-    model_dir_fused = f"{source_model_dir}-fused"
-
-    # this combines adapter with the original model to produce the updated model
     try:
-        fine_tune(
-            model=source_model_dir,
-            save_path=model_dir_fused,
-            adapter_file=adapter_file,
-            de_quantize=not skip_de_quantize,
-        )
-    except (requests_exceptions.HTTPError, hf_errors.HFValidationError) as e:
-        click.secho(
-            f"Failed to fine tune: {e}",
-            fg="red",
-        )
-        raise click.exceptions.Exit(1)
-
-    model_dir_fused_pt = f"{model_name}-trained"
-    # this converts MLX to PyTorch
-    convert_between_mlx_and_pytorch(
-        hf_path=model_dir_fused, mlx_path=model_dir_fused_pt, local=True, to_pt=True
-    )
-
-    logger.info(f"deleting {model_dir_fused}...")
-    shutil.rmtree(model_dir_fused)
-
-    convert_llama_to_gguf(
-        model=Path(model_dir_fused_pt),
-        pad_vocab=True,
-        skip_unknown=True,
-        outfile=f"{model_dir_fused_pt}/{model_name}.gguf",
-    )
-
-    logger.info(f"deleting safetensors files from {model_dir_fused_pt}...")
-    for file in glob(os.path.join(model_dir_fused_pt, "*.safetensors")):
-        os.remove(file)
-
-    # quantize to 4-bit GGUF (optional)
-    if not skip_quantize:
-        gguf_model_dir = f"{model_dir_fused_pt}/{model_name}.gguf"
-        gguf_model_q_dir = f"{model_dir_fused_pt}/{model_name}-Q4_K_M.gguf"
-        run_quantize(gguf_model_dir, gguf_model_q_dir, "Q4_K_M")
-
-    logger.info(f"deleting {model_dir_fused_pt}/{model_name}.gguf...")
-    os.remove(os.path.join(model_dir_fused_pt, f"{model_name}.gguf"))
-
-    logger.info(f"deleting {source_model_dir}...")
-    shutil.rmtree(source_model_dir)
-    click.echo("ᕦ(òᴗóˇ)ᕤ Model convert completed successfully! ᕦ(òᴗóˇ)ᕤ")
+        src_model_fmt = get_model_format(model)
+        converter_cls = Converter.get_fmt_converter(src_model_fmt, target_format)
+        converter_instance = converter_cls(model, destination, outtype)
+        converter_instance.convert()
+        logger.info("ᕦ(òᴗóˇ)ᕤ Model convert completed successfully! ᕦ(òᴗóˇ)ᕤ")
+    except ValueError as e:
+        raise ValueError(f"Model conversion failed: {e}") from e
+
+
+def get_model_format(model: Path) -> SUPPORTED_FORMATS:
+    if is_model_safetensors(model):
+        return SUPPORTED_FORMATS.SAFETENSORS
+
+    if is_model_gguf(model):
+        return SUPPORTED_FORMATS.GGUF
+
+    raise ValueError(
+        f"Source model {model} is in an unsupported format. Supported formats are {SUPPORTED_FORMATS._member_names_}"
+    ) from None
```

Comment:
Can we add unit testing for this function as well?

Reply:
added